### DISCUSSION TITLE
Is the handle_tasks system in the async_comput example polling or blocking?

### DISCUSSION THREAD

**Discussion by rohel01** [bevyengine/bevy#8085] [Q&A] [answered]
Hello,

I am trying to understand the async_comput example, specifically the lines below:

```Rust
) {
    for (entity, mut task) in &mut transform_tasks {
        if let Some(transform) = future::block_on(future::poll_once(&mut task.0)) {
```

Since we are in a system, I would expect this code to do a non blocking poll, but the `future::block_on` seems to indicate otherwise. Is it blocking?

Kind regards,

**Comment by Aiden2207** [comment, marked as answer]
It doesn't block, `future::poll_once` means that the future will complete immediately when `.await`ed or `block_on`ed, assuming the future is well-behaved. The code here just checks to see if the future is done, and consumes the output if it is.

**Comment by rohel01** [comment]
Ok, this is what I expected.
Thank you for the clarification.
